libkbfs/key_manager.go

```diff
@@ -124,19 +124,80 @@ func (km *KeyManagerStandard) getTLFCryptKey(ctx context.Context,
 		return TLFCryptKey{}, err
 	}
 
-	localMakeRekeyReadError := func() error {
-		return makeRekeyReadError(ctx, km.config, kmd, keyGen, uid, username)
+	clientHalf, serverHalfID, cryptPublicKey, err :=
+		km.getTLFCryptKeyParams(ctx, kmd, keyGen, uid, username, flags)
+
+	var notPerDeviceEncrypted bool
+	if _, notPerDeviceEncrypted = err.(TLFCryptKeyNotPerDeviceEncrypted); notPerDeviceEncrypted {
+		// get the key we want using the current crypt key
+		currKeyGen := kmd.LatestKeyGeneration()
+		// look in the cache first
+		tlfCryptKey, err = kcache.GetTLFCryptKey(tlfID, currKeyGen)
+		switch err := err.(type) {
+		case nil:
+			break
+		case KeyCacheMissError:
+			// not cached, look up the params
+			var err2 error
+			clientHalf, serverHalfID, cryptPublicKey, err2 =
+				km.getTLFCryptKeyParams(ctx, kmd, currKeyGen, uid, username, flags)
+			if err2 != nil {
+				return TLFCryptKey{}, err2
+			}
+			break
+		default:
+			return TLFCryptKey{}, err
+		}
+	} else if err != nil {
+		return TLFCryptKey{}, err
+	}
+
+	if (tlfCryptKey == TLFCryptKey{}) {
```

Comment:
...in particular, this only applies for `latestKey`, right? Shouldn't it be put in the cache miss case above?